app/controllers/spree/admin/orders_controller_decorator.rb

```diff
@@ -84,17 +85,35 @@ def resend
   end
 
   def invoice
-    template = if Spree::Config.invoice_style2? then "spree/admin/orders/invoice2" else "spree/admin/orders/invoice" end
-    pdf = render_to_string pdf: "invoice-#{@order.number}.pdf", template: template, formats: [:html], encoding: "UTF-8"
+    pdf = render_to_string pdf: "invoice-#{@order.number}.pdf",
+                           template: invoice_template,
+                           formats: [:html], encoding: "UTF-8"
+
     Spree::OrderMailer.invoice_email(@order.id, pdf).deliver
     flash[:success] = t('admin.orders.invoice_email_sent')
 
     respond_with(@order) { |format| format.html { redirect_to edit_admin_order_path(@order) } }
   end
 
+  def bulk_invoice
+    orders = params[:order_ids]
+
+    combined_pdf = CombinePDF.new
+
+    orders.each do |order_id|
+      @order = Spree::Order.find(order_id)
```

Comment:
we should rather fetch all the orders in one shot to save a considerable amount of round-trips to the DB.